File: android/app/src/main/java/com/kurt/bgmate/data/remote/llm/LlmRequest.kt
package com.kurt.bgmate.data.remote.llm

data class LlmMessage(val role: String, val content: String)

data class LlmRequest(
    val messages: List<LlmMessage>,
    val systemPrompt: String? = null,
    val maxTokens: Int = 4096
)

File: flutter/lib/data/remote/ai/llm_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'llm_request.freezed.dart';

@freezed
sealed class LlmRequest with _$LlmRequest {
  const factory LlmRequest({
    required List<LlmMessage> messages,
    @Default(null) String? systemPrompt,
    @Default(4096) int maxTokens,
  }) = _LlmRequest;
}

@freezed
sealed class LlmMessage with _$LlmMessage {
  const factory LlmMessage({required String role, required String content}) =
      _LlmMessage;
}
